akka-actor-typed/src/main/scala/akka/actor/typed/SupervisorStrategy.scala

```diff
@@ -197,6 +200,56 @@ sealed abstract class SupervisorStrategy {
   def withLoggingEnabled(on: Boolean): SupervisorStrategy
 }
 
+sealed abstract class RestartSupervisorStrategy extends SupervisorStrategy {
+
+  /**
+   * Scala API: Restart with a limit of number of restart retries.
+   * The number of restarts are limited to a number of restart attempts (`maxNrOfRetries`)
+   * within a time range (`withinTimeRange`). When the time window has elapsed without reaching
+   * `maxNrOfRetries` the restart count is reset.
+   *
+   * The strategy is applied also if the actor behavior is deferred and throws an exception during
+   * startup.
+   *
+   * @param maxNrOfRetries the number of times a child actor is allowed to be restarted,
+   *   if the limit is exceeded the child actor is stopped
+   * @param withinTimeRange duration of the time window for maxNrOfRetries
+   */
+  def withLimit(maxNrOfRetries: Int, withinTimeRange: FiniteDuration): RestartSupervisorStrategy
+
+  /**
+   * Java API: Restart with a limit of number of restart retries.
+   * The number of restarts are limited to a number of restart attempts (`maxNrOfRetries`)
+   * within a time range (`withinTimeRange`). When the time window has elapsed without reaching
+   * `maxNrOfRetries` the restart count is reset.
+   *
+   * The strategy is applied also if the actor behavior is deferred and throws an exception during
+   * startup.
+   *
+   * @param maxNrOfRetries the number of times a child actor is allowed to be restarted,
+   *   if the limit is exceeded the child actor is stopped
+   * @param withinTimeRange duration of the time window for maxNrOfRetries
+   */
+  def withLimit(maxNrOfRetries: Int, withinTimeRange: java.time.Duration): RestartSupervisorStrategy
+
+  /**
+   * Stop or keep child actors when the parent actor is restarted.
+   * @param enabled if `true` then child actors are stopped, otherwise they are kept
+   */
+  def withStopChildren(enabled: Boolean): RestartSupervisorStrategy
```

Comment:
how about `StopAll` and `StopNone`?

Reply:
Seems unlikely we will add more states so boolean is fine.

Adding a note about the default would be good though.